internal/cli/managed.go

```diff
@@ -0,0 +1,23 @@
+package cli
+
+import (
+	"errors"
+
+	"github.com/kyma-project/cli/internal/clusterinfo"
+)
+
+const managedKymaWarning = "CAUTION: You are trying to use Kyma CLI to change a managed Kyma runtime (SAP Kyma Runtime). This action may corrupt the Kyma runtime. Proceed at your own risk."
+
+// DetectManagedEnvironment introduces a step that checks if the target runtime is a managed Kyma runtime. CLI should be used with caution in such environment and user is prompted for confirmation.
+func DetectManagedEnvironment(cmd *Command) error {
+	detectStep := cmd.NewStep("Detecting managed Kyma runtime")
```

Comment:
IMHO not everything that happens on the process needs to be added as a step. Steps should be well thought out and provide useful feedback to the user. This will in most cases pollute the output and in the cases it is relevant, the prompt already provides all the information.

With that change, it is enough to provide the current step to the function and not the whole command.

Reply:
The problem is, we don't have any "current step" at the time this function is invoked... I'll take a look at what we can do.

Reply:
I see, in that case we can pass an empty step and only use it to prompt.

Reply:
👍